### DISCUSSION TITLE
Including semmle libraries in custom query suites

### DISCUSSION THREAD

**Discussion by mutasDev** [github/codeql#10901] [Q&A] [answered]
Hey,

I am trying to create a custom query set for python (and in future some further languages) and for those I need to include some experimental libraries from semmle, more specifically , e.g. : `experimental.semmle.python.Concepts`.

I tried copying the complete experimental folder from the codeql repository into the folder where the query lies, but that does not seem to work.

How can I include tertiary libraries into own suites?

**Comment by tausbn** [comment, marked as answer]
May I ask how you're creating your query pack? For me, the following interaction with the CodeQL CLI seemed to work just fine:

```
% codeql pack init test
Created qlpack at .../test/qlpack.yml
% cd test
% codeql pack add codeql/python-queries
Adding codeql/python-queries@0.5.1 to dependencies.
Saving qlpack.yml.
Reinstalling dependencies.
Package install location: ~/.codeql/packages
Installed fresh codeql/python-queries@0.5.1
Installed fresh codeql/python-all@0.6.1 (library)
Installed fresh codeql/suite-helpers@0.1.0
```
At this point I added a file `test.ql` with the following contents:
```ql
import python
import experimental.semmle.python.Concepts

select 1
```
and now compiling it works:
```
% codeql query compile test.ql
Compiling query plan for .../test/test.ql.
Done [1/1 comp 3.4s] .../test/test.ql.
```

So basically I _think_ what may be missing here is the `codeql/python-queries` dependency. Once that has been installed, everything should work, hopefully. 🤞
